src/linear_map.rs

```diff
@@ -186,14 +187,17 @@ where
     /// assert_eq!(map.insert(37, "c").unwrap(), Some("b"));
     /// assert_eq!(map[&37], "c");
     /// ```
-    pub fn insert(&mut self, key: K, mut value: V) -> Result<Option<V>, (K, V)> {
+    pub fn insert(&mut self, key: K, mut value: V) -> CapacityResult<Option<V>, (K, V)> {
         if let Some((_, v)) = self.iter_mut().find(|&(k, _)| *k == key) {
             mem::swap(v, &mut value);
-            return Ok(Some(value));
+            return CapacityResult::ok(Some(value))
         }
 
-        self.buffer.push((key, value))?;
-        Ok(None)
+        let p = self.buffer.push((key, value));
+        if p.is_err() {
+            return p.map(|_| None); // convert type
```

Comment:
Maybe this can be made more ergonomic by implementing `From<CapacityResult<T, R>> for CapacityResult<T2, R2> where T2: From<T>, R2: From<R>`